_posts/2023-08-30-industrial-computers-for-koji.md

```diff
@@ -0,0 +1,159 @@
+
+1. Koyo Click C2-01CPU: Click Plus Ethernet - **\$97 USD** 
+2. Koyo Click C2-08D2-6C: I/O module with 4x DIs, 4x DOs, 4x AIs (4-20mA), and 2x AOs (4-20mA) - **$90 USD**
+3. Koyo Click C0-04THM: Input module with 4x TCs - **$217 USD** (TC/RTD modules are so expensive...)
+
+![](/assets/images/koji_plc1/plc1.JPG){:width="450" .center}
+
+Bonus: programming software for Koyo Click is free!
+
+Thank you, Automation Direct (they are not a sponsor of this post). 
+
+Since this project is mostly 'for fun', we can also also get away with refurbished or used industrial components. Our target budget for this project will be around \$1200 CAD - like your average home PC build, and not out of reach for small scale koji/natto/mushroom makers. 
+
+After purchasing the necessary DIN rails, breakers, fuses, power supplies, terminal blocks, and relays, we can set up our system on a mock "test bench":
+
+![](/assets/images/koji_plc1/testbench1.JPG){:width="450" .center}
+
+Here's where you can really appreciate the convenience of industrial control panels. Everything here is mounted on a DIN rail, a piece of steel channel 35mm wide that allows users to switch out any component as needed. All wires are terminated at screw-in terminal blocks, meaning things can be re-wired quickly without soldering, easily diagnosed with a multimeter, but still be secure enough not to come loose from vibrations. 
+
+On the back of the test bench, we'll mount our sensors. These would normally go into the 'field', in our case, the koji room. 
+
+![](/assets/images/koji_plc1/testbench2.JPG){:width="450" .center}
+
+<div class="callout">
+⚠ We are dealing with 120VAC line voltage, which can be dangerous. If you plan on DIY-ing something based on this post, please make sure you know what you're doing, or get a competent electrician. Also, there are codes that you must follow if you're actually putting together an industrial panel. This post is mostly for 'educational' purposes - my intent isn't to tell you how to put together a panel, but how PLCs work at a basic level. Stay safe!
+</div>
+
+
+## Next Steps and Notes
+
+That's it for now! I'll do some basic testing on this test bench before transferring it to a proper electrical enclosure, with some additional functionality. 
+
+I'll be using this PLC to control a "general purpose" solid-state fermentation room, in order to produce time-dependent temperature/RH/CO2 profiles for koji, natto, and mushroom growing. 
+
+This post barely scratches the surface of the capabilities of PLCs. I won't be going into any wiring details, electrical connections, how relays work, etc. If you want to learn more, see the 'References' section below. 

```

Comment:
I think I would like to know more about how you built this set up (speaking on behalf of my self and the audience). Is that for a future post?

Reply:
Hmm that might be a good idea, I might save it when I end up throwing this thing into panel